src/identity_committer.rs

```diff
@@ -0,0 +1,204 @@
+use crate::{
+    app::{Hash, TreeState},
+    contracts::Contracts,
+    database::Database,
+};
+use eyre::eyre;
+use std::sync::{atomic::Ordering, Arc};
+use tokio::{
+    select,
+    sync::{mpsc, mpsc::error::TrySendError, RwLock},
+    task::JoinHandle,
+};
+use tracing::{debug, error, info, instrument};
+
+struct RunningInstance {
+    #[allow(dead_code)]
+    handle:          JoinHandle<eyre::Result<()>>,
+    wake_up_sender:  mpsc::Sender<()>,
+    shutdown_sender: mpsc::Sender<()>,
+}
+
+impl RunningInstance {
+    fn wake_up(&self) -> eyre::Result<()> {
+        // We're using a 1-element channel for wake-up notifications. It is safe to
+        // ignore a full channel, because that means the committer is already scheduled
+        // to wake up and will process all requests inserted in the database.
+        match self.wake_up_sender.try_send(()) {
+            Ok(_) => {
+                debug!("Scheduled a committer job.");
+                Ok(())
+            }
+            Err(TrySendError::Full(_)) => {
+                debug!("Committer job already scheduled.");
+                Ok(())
+            }
+            Err(TrySendError::Closed(_)) => Err(eyre!("Committer thread terminated unexpectedly.")),
+        }
+    }
+
+    async fn shutdown(self) -> eyre::Result<()> {
+        info!("Sending a shutdown signal to the committer.");
+        // Ignoring errors here, since we have two options: either the channel is full,
+        // which is impossible, since this is the only use, and this method takes
+        // ownership, or the channel is closed, which means the committer thread is
+        // already dead.
+        self.shutdown_sender.send(()).await.ok();
```

Comment:
`.ok()` just returns an `Option<T>` containing the value, if any. If we want to dispose of the result without triggering a panic or the `must_use` lint the pattern is:

```rust
let _ = self.shutdown_sender.send(()).await;
```